src/cpio/client_providers/private_key_fetcher_provider/azure/azure_private_key_fetcher_provider_utils.h

```diff
@@ -31,7 +203,69 @@ class AzurePrivateKeyFetchingClientUtils {
   static void CreateHttpRequest(
       const PrivateKeyFetchingRequest& private_key_fetching_request,
       core::HttpRequest& http_request);
+
+  /**
+   * @brief Generate a new wrapping key
+   */
+  static std::pair<std::shared_ptr<EvpPkeyWrapper>,
+                   std::shared_ptr<EvpPkeyWrapper>>
+  GenerateWrappingKey();
+
+  /**
+   * @brief Convert a wrapping key in PEM
+   *
+   * @param wrappingKey RSA public key used to wrap a key.
+   */
+  static std::string EvpPkeyToPem(EVP_PKEY* wrappingKey);
+
+  /**
+   * @brief Convert a public PEM wrapping key to pkey
+   *
+   * @param wrappingPemKey RSA PEM key used to wrap a key.
+   */
+  static EVP_PKEY* GetPublicEvpPkey(std::string wrappingPemKey);
+
+  /**
+   * @brief Convert a private PEM wrapping key to pkey
+   *
+   * @param wrappingPemKey RSA PEM key used to wrap a key.
+   */
+  static EVP_PKEY* GetPrivateEvpPkey(std::string wrappingPemKey);
+
+  /**
+   * @brief Generate hex hash on wrapping key
+   */
+  static std::string CreateHexHashOnKey(EVP_PKEY* publicKey);
```

Comment:
Similarly to the comment on GetPublicEvpPkey , we can use `static std::string CreateHexHashOnKey(std::shared_ptr<EvpPkeyWrapper> publicKey);`. And there are some other functions that you can make the same changes

Reply:
Will be solved by this issue: https://github.com/microsoft/privacy-sandbox-dev/issues/148